core/unit_test/TestViewTypedefs.cpp

```diff
@@ -0,0 +1,274 @@
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+  static_assert(std::is_same_v<typename ViewType::uniform_runtime_nomemspace_type,
+                               Kokkos::View<typename data_analysis<DataType>::runtime_data_type, uniform_layout_type,
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+  static_assert(std::is_same_v<typename ViewType::uniform_runtime_const_nomemspace_type,
+                               Kokkos::View<typename data_analysis<DataType>::runtime_const_data_type, uniform_layout_type,
+                                            anonymous_device_type, Kokkos::MemoryTraits<0>>>);
+
+
+  // ==================================
+  // mdspan compatibility
+  // ==================================
+
+  // FIXME: This typedef caused some weird issue with MSVC+NVCC
+  // static_assert(std::is_same_v<typename ViewType::layout_type, Layout>);
+  // FIXME: Not supported yet
+  // static_assert(std::is_same_v<typename ViewType::extents_type, >);
+  // static_assert(std::is_same_v<typename ViewType::mapping_type, >);
+  // static_assert(std::is_same_v<typename ViewType::accessor_type, >);
+
+  static_assert(std::is_same_v<typename ViewType::element_type, ValueType>);
+  // FIXME: should be remove_const_t<element_type>
+  static_assert(std::is_same_v<typename ViewType::value_type, ValueType>);
+  // FIXME: should be extents_type::index_type
+  static_assert(std::is_same_v<typename ViewType::index_type, typename Space::memory_space::size_type>);
+  // FIXME: this isn't given in View since for example SYCL has "int" as its size_type
+  // static_assert(std::is_same_v<typename ViewType::size_type, std::make_unsigned_t<typename ViewType::index_type>>);
+  static_assert(std::is_same_v<typename ViewType::rank_type, size_t>);
+
+  // FIXME: should come from accessor_type
+  static_assert(std::is_same_v<typename ViewType::data_handle_type, typename ViewType::pointer_type>);
+  static_assert(std::is_same_v<typename ViewType::reference, typename ViewType::reference_type>);
+  return true;
+};
+
+// Helper function to unpack data type and other args from the View, and pass them on
+template<class T, class ... ViewArgs>
+struct ViewParams {};
+
+template<class L, class S, class M, class HostMirrorSpace, class ValueType, class ReferenceType, class T, class ... ViewArgs>
+constexpr bool test_view_typedefs(ViewParams<T, ViewArgs...>) {
+  return test_view_typedefs_impl<Kokkos::View<T, ViewArgs...>, Kokkos::ViewTraits<T, ViewArgs...>,
+                                 T, L, S, M, HostMirrorSpace, ValueType, ReferenceType>();
+}
+
+
+constexpr bool is_host_exec = std::is_same_v<Kokkos::DefaultExecutionSpace, Kokkos::DefaultHostExecutionSpace>;
+
+#if defined(KOKKOS_ENABLE_CUDA_UVM) || defined(KOKKOS_ENABLE_IMPL_CUDA_UNIFIED_MEMORY) || defined(KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY)
```

Comment:
Is this better than checking against `KOKKOS_HAS_SHARED_SPACE`?
Is `SYCLSharedUSMSpace` not applicable for this?

Reply:
This isn't the same. The condition here is needed when DefaultExecutionSpace::memory_space is host accessible, but DefaultExecutionSpace is not DefaultHostExecutionSpace. I.e. it only applies if we use the deprecated KOKKOS_ENABLE_CUDA_UVM, or one of the new UNIFIED_MEMORY build options.

Reply:
This is in principle orthogonal to the question of whether Kokkos::SharedSpace is a thing.